server/src/routes/auth/signUp/signUp.js

```diff
@@ -1,10 +1,35 @@
+import bcrypt from 'bcryptjs';
+import User from '../../../models';
+
 /**
  * @param {import('express').Request} req
  * @param {import('express').Response} res
  */
+
 const signUp = async (req, res) => {
-  const { email } = req.body;
-  res.json({ email });
+  const { username, email, password } = req.body;
+
+  // checking if this email is already in db
+  const emailExist = await User.findOne({ email });
+  if (emailExist)
+    return res.status(400).send({ status: 'fail', message: `An account for email ${email} already exist` });
+
+  // hash the password
+  const salt = await bcrypt.genSalt(10);
+  const pswHash = await bcrypt.hash(password, salt);
```

Comment:
I would suggest to avoid contractions in variable names. We are saving only few keystrokes and giving up a lot on readability of the code.